code/modules/reagents/reagent_dispenser.dm

```diff
@@ -65,7 +65,10 @@
 	reagent_id = /datum/reagent/fuel
 
 /obj/structure/reagent_dispensers/fueltank/boom()
-	explosion(get_turf(src), 0, 1, 5, flame_range = 5)
+	var/light_explosion_range = CLAMP(round(reagents.get_reagent_amount(/datum/reagent/fuel)/200, 1), 1, 5) //explosion range should decrease when there is less fuel in the tank
+	var/flame_explosion_range = CLAMP(light_explosion_range + 1, 1, 5) //Fire explosion is always one bigger than light explosion
```

Comment:
Is clamp even needed here? maximum amount is 1000 and 1000/200 is 5.

Reply:
yeah then first clamp is needed but second one will receive minimal value of 2.

Reply:
yes but I don't want the fire explosion to be above 5 (when welder fuel tank is full)

Reply:
Thinking about it now I should make it so the flame_explosion is = CLAMP(light_explosion, 2, 5)

Reply:
Keep the clamping before some wack circuit bug comes a long that allows you to nuke the station with welding tanks

Reply:
So I'll just change it to flame_explosion is = CLAMP(light_explosion, 2, 5) so it looks better